### PR TITLE
feat(js): inbox support multiple counts for the provided filters

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?
In this PR:
- API changes in the `GET /notifications/counts` endpoint which supports multiple filters in query params as JSON object
- JS SDK changes that reflect the API functionality with better interface

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/inbox/dtos/get-notifications-count-request.dto.ts [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
 }
+
+export class GetNotificationsCountRequestDto {
+  @IsDefined()
+  @Transform(({ value }) => {
+    try {
+      const filters = JSON.parse(value);
+      if (Array.isArray(filters)) {
+        return filters.map((el) => plainToClass(NotificationsFilter, el));
+      }
```
Comment (LetItRock) [inline apps/api/src/app/inbox/dtos/get-notifications-count-request.dto.ts:28]: The filters are sent in the query params as the stringified JSON object. We need to parse it back from string to object and run the validation on its fields.

File: packages/js/src/feeds/feeds.ts [hunk 2]
```diff
@@ -43,28 +48,30 @@ export class Feeds extends BaseModule {
     });
   }
 
-  async fetchCount(countArgs: FetchCountArgs = {}): Promise<FetchCountResponse> {
+  async fetchCount(args?: FetchFilterCountArgs): Promise<FetchFilterCountResponse>;
+  async fetchCount(args?: FetchFiltersCountArgs): Promise<FetchFiltersCountResponse>;
+  async fetchCount(args: FetchCountArgs): Promise<FetchCountResponse> {
```
Comment (LetItRock) [inline packages/js/src/feeds/feeds.ts:53]: the supported calls are like:
```
// get all unread/read, archived/unarchived notifications counts
const countResponse = await novu.feeds.fetchCount(); 

// get unread notifications counts
const countResponse = await novu.feeds.fetchCount({ read: false }); 

// get unread counts and archived counts with tag newsletter
const countResponse = await novu.feeds.fetchCount({
   filters: [{ read: false }, { tags: ["newsletter"], archived: false }]
}); 

```

File: packages/js/src/feeds/feeds.ts [hunk 3]
```diff
@@ -43,28 +48,30 @@ export class Feeds extends BaseModule {
...
-        return response;
+        return args && 'filters' in args ? response : { data: response.data[0] };
       } catch (error) {
         this._emitter.emit('feeds.fetch_count.error', { args, error });
         throw error;
       }
     });
   }
 
+  async read(args: BaseArgs): Promise<Notification>;
```
Comment (LetItRock) [inline packages/js/src/feeds/feeds.ts:74]: It was missing

File: packages/js/src/ui/api/hooks/useFetchUnreadCount.ts [hunk 4]
```diff
@@ -1,28 +1,31 @@
...
 
 export const useFetchUnreadCount = ({ filters, onSuccess, onError }: UseUnreadCountOptions = { filters: {} }) => {
   const novu = useNovu();
 
-  const [unreadCount, { refetch }] = createResource({ ...filters, read: false }, async (payload?: FetchCountArgs) => {
-    try {
-      const response = await novu.feeds.fetchCount(payload);
-      const count = response.data.count;
-      onSuccess?.(count);
+  const [unreadCount, { refetch }] = createResource(
```
Comment (LetItRock) [inline packages/js/src/ui/api/hooks/useFetchUnreadCount.ts:14]: this hook is not used but I had to update it's code

File: apps/api/src/app/inbox/dtos/get-notifications-count-request.dto.ts (hunk 1 above)
Comment (SokratisVidros) [reply apps/api/src/app/inbox/dtos/get-notifications-count-request.dto.ts:28]: The stringified JSON filters in counts creates a discrepancy between the count and the notifications endpoint where one filter is specified via separate query string parameters.

I don't have a good suggestion yet but I suggest we add a task to revise this post-release and opt-in for more alignment between the two endpoints.

File: apps/api/src/app/inbox/usecases/notifications-count/notifications-count.usecase.ts [hunk 5]
```diff
@@ -34,19 +36,20 @@ export class NotificationsCount {
...
-    const filter = { tags: command.tags, read: command.read, archived: command.archived };
-    const count = await this.messageRepository.getCount(
-      command.environmentId,
-      subscriber._id,
-      ChannelTypeEnum.IN_APP,
-      filter,
-      { limit: MAX_NOTIFICATIONS_COUNT }
+    const getCountPromises = command.filters.map((filter) =>
+      this.messageRepository.getCount(command.environmentId, subscriber._id, ChannelTypeEnum.IN_APP, filter, {
+        limit: MAX_NOTIFICATIONS_COUNT,
```
Comment (SokratisVidros) [inline apps/api/src/app/inbox/usecases/notifications-count/notifications-count.usecase.ts:46]: 💯

File: packages/js/src/feeds/feeds.ts (hunk 2 above)
Comment (SokratisVidros) [reply packages/js/src/feeds/feeds.ts:53]: The parameters work really nicely. Should we rename `fetchCount` to just `count` in a more ORM like fashion?

File: packages/js/src/ui/api/hooks/useFetchUnreadCount.ts [hunk 6]
```diff
@@ -1,28 +1,31 @@
...
+        const count = response.data[0].count;
+        onSuccess?.(count);
 
-      return count;
-    } catch (error) {
-      console.warn('Error fetching unread count:', error);
-      onError?.(error);
+        return count;
+      } catch (error) {
+        console.warn('Error fetching unread count:', error);
```
Comment (SokratisVidros) [inline packages/js/src/ui/api/hooks/useFetchUnreadCount.ts]: Let's remove the console.warn and just keep the onError handler.

**Review by SokratisVidros — APPROVED** [review]

File: apps/api/src/app/inbox/dtos/get-notifications-count-request.dto.ts (hunk 1 above)
Comment (LetItRock) [reply apps/api/src/app/inbox/dtos/get-notifications-count-request.dto.ts:28]: ok, I'll add one

**Pull request merged**